cedar-policy-validator/src/entity_manifest.rs

```diff
@@ -538,8 +538,19 @@ fn entity_manifest_from_expr(
             .union(&entity_manifest_from_expr(right)?.empty_paths())),
         ExprKind::UnaryApp { op, arg } => {
             match op {
-                // both unary ops are on booleans, so they are simple
+                // these unary ops are on primitive types, so they are simple
                 UnaryOp::Not | UnaryOp::Neg => Ok(entity_manifest_from_expr(arg)?.empty_paths()),
+                UnaryOp::IsEmpty => {
+                    // PANIC SAFETY: Typechecking succeeded, so type annotations are present.
+                    #[allow(clippy::expect_used)]
+                    let ty = arg
+                        .data()
+                        .as_ref()
+                        .expect("Expected annotated types after typechecking");
```

Comment:
I don't understand why we do this - it's just a sanity check?

Reply:
I don't understand any of this `entity_manifest.rs` code.  I was trying to take the relevant code from `BinaryApp` and adapt it in a sane way for `UnaryApp`

Reply:
I don't understand either, but this looks reasonable to me

Reply:
Yeah, looks good assuming the `BinaryApp` case is fine.